sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -199,7 +199,12 @@ case class Divide(left: Expression, right: Expression) extends BinaryArithmetic
 
   override def symbol: String = "/"
   override def decimalMethod: String = "$div"
-  override def nullable: Boolean = true
+  override def nullable: Boolean = (left.nullable ||
+    !(!right.nullable && right.isInstanceOf[Literal] &&
+      ((right.asInstanceOf[Literal].value.isInstanceOf[Decimal] &&
```

Comment:
Since I do not have the benchmark that whose hotspot is related to them, I removed them in ```Divide``` and ```Remainder```.